Add Uniswap V3 LP Bridge to Bridges

ndebrouwer opened the pull request:
Uniswap V3 LP Bridge, w/ front-end bridge data classes as well. Implemented getAuxDataLP function as suggested by Joe. 



Note regarding a bug: yarn test:client will only compile when the hardhat config does not specify the compiler version otherwise it causes "Variable headStart is 1 slot(s) too deep inside the stack." error even if the stack is not full. In order to get the typechain types generated / produce hardhat compile success, I changed the hardhat config temporarily. I changed it back for the purposes of this PR to avoid conflicts with master branch, but to get the yarn test:client to run the hardhat config needs to be changed at least temporarily so the typechain types can be generated. 

In essence;
 solidity: {
    compilers: [
      {
        version: '0.8.10',
      },
    ],
    settings: {
      optimizer: { enabled: true, runs: 200 },
    },
  },

this above fails
while this below compiles fine. 
solidity: {
    version: "0.8.10",
    settings: {
      optimizer: { enabled: true, runs: 200 },
    },
  },

joeandrews commented:
Hi, please can you make the above change to the hardhat.config, and rebase from Master to check that all of the other tests still pass?

Thanks!

Joe

joeandrews commented on src/test/example/Example.t.sol:
```diff
@@ -1,118 +0,0 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity 0.8.10;
```
Can you bring this file back?

joeandrews commented on src/bridges/example/ExampleBridge.sol:
```diff
@@ -1,57 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-only
-// Copyright 2020 Spilsbury Holdings Ltd
```
This file needs to remain in the repo as an example.

joeandrews commented on README.md:
```diff
@@ -1,5 +1,7 @@
 [![CircleCI](https://circleci.com/gh/AztecProtocol/aztec-connect-bridges/tree/master.svg?style=shield)](https://circleci.com/gh/AztecProtocol/aztec-connect-bridges/tree/master)
 
+# Uniswap V3 Bridge for Liquidity Provision
```
Remove

joeandrews commented on package.json:
```diff
@@ -9,14 +9,14 @@
     "test": "yarn test:contracts && yarn test:clients",
     "test:client": "jest test",
-    "test:contracts": "forge test -vvv",
+    "test:contracts": "forge test -vv",
```
Why has this changed?

ndebrouwer commented:
Hey Joe, I rebased from master as you instructed, changed the hardhat config and everything still runs and both yarn test:client and yarn test:contracts pass. I also added the files that were missing via the rebase. Everything you mentioned should (?) be addressed but if there's something else you noticed just comment and I will take care of it.

Additionally I made some minor changes which are inconsequential but which I feel compelled to note: I added the uint24 type to the SolidityType enum in bridge-data.ts. I also added the assume function in Vm.sol so that the assume cheatcode for foundry works.

ndebrouwer commented:
Hey Joe, I rebased from master as you instructed, changed the hardhat config and everything still runs and both yarn test:client and yarn test:contracts pass. I also added the files that were missing via the rebase. Everything you mentioned should (?) be addressed but if there's something else you noticed just comment and I will take care of it. Additionally I made some minor changes which are inconsequential but which I feel compelled to note: I added the uint24 type to the SolidityType enum in bridge-data.ts. I also added the assume function in Vm.sol so that the assume cheatcode for foundry works.
